src/query/service/src/pipelines/executor/pipeline_pulling_executor.rs

```diff
@@ -184,14 +186,25 @@ impl PipelinePullingExecutor {
                     continue;
                 }
                 Err(_disconnected) => {
+                    let mut killed = false;
+
                     if !self.executor.is_finished() {
+                        killed = true;
                         self.executor.finish(None);
                     }
 
                     self.state.wait_finish();
 
                     if self.state.is_catch_error() {
-                        return Err(self.state.get_catch_error());
+                        let error_code = self.state.get_catch_error();
+
+                        // abort errors are caused by interrupts and should be ignored.
```

Comment:
I don't understand the comment.

1. By interrupting and ? should be ignored.
2. Why was the SQL in issue aborted?  (Server wasn't shot down and the query wasn't killed)

Reply:
for 'select * from xx limit 20', when 20 rows of data are returned, the query should have been finished. since the processor cannot timely known that, we try to kill the query in here.

Reply:
Updated comment.